model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -54,6 +68,34 @@ cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_predicti
     }
     return hard_prediction;
 }
+
+std::vector<Annotation> create_annotation_from_segmentation_map(const cv::Mat& hard_prediction, const cv::Mat& soft_prediction, std::vector<std::string> label_map) {
+    std::vector<Annotation> annotations = {};
+    cv::Mat label_index_map;
+    cv::Mat current_label_soft_prediction;
+    size_t labelID = 0;
+    for (int index = 1; index < soft_prediction.channels(); index++) {
```

Comment:
If `1 == soft_prediction.channels()`, empty vector is returned. Is this intentional?

Reply:
Correct, first label is a background and not relevant for the contours.

Reply:
Please throw if someone calls `getContours()` and `imageResult->soft_prediction.channels()` is 1

Reply:
This is still applicable